sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -163,7 +163,7 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
 
 }
 
-private[csv] class CsvOutputWriter(
```

Comment:
Please use `private[sql]` instead of removal.

Reply:
If you don't mind, can we have this as a separate file?

Reply:
I think we can just make it public since the execution package is meant to be internal. See the comment https://github.com/apache/spark/pull/24005#pullrequestreview-212555802

I will put it into a separate file.

Reply:
For `XXXOutputWriter`, we usually keep them `private[sql]`.

Reply:
`private[xxx]` yes, but if that has to be broader than `private[sql]` or `private[spark]`, it actually we're partially reverting SPARK-16964.

Reply:
Got it~ I missed the point, @HyukjinKwon and @gengliangwang .

Reply:
Thanks for pointing this out - I had to remain it for me by myself.